src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -4398,6 +4398,86 @@ public void ToString_AbstractRecord()
             CompileAndVerify(comp, expectedOutput: "C2 { I1 = 42, I2 = 43 }", verify: Verification.Skipped /* init-only */);
         }
 
+        [Fact, WorkItem(47672, "https://github.com/dotnet/roslyn/issues/47672")]
+        public void ToString_RecordWithIndexer()
+        {
+            var src = @"
+var c1 = new C1(42);
+System.Console.Write(c1.ToString());
+
+record C1(int I1)
+{
+    private int field = 44;
+    public int this[int i] => 0;
+    public int PropertyWithoutGetter { set { } }
+    public int P2 { get => 43; }
+    public ref int P3 { get => ref field; }
+    public event System.Action a;
+
+    private int field1 = 100;
+    internal int field2 = 100;
+    protected int field3 = 100;
+    private protected int field4 = 100;
+    internal protected int field5 = 100;
+
+    private int Property1 { get; set; } = 100;
+    internal int Property2 { get; set; } = 100;
+    protected int Property3 { get; set; } = 100;
+    private protected int Property4 { get; set; } = 100;
+    internal protected int Property5 { get; set; } = 100;
+}
+";
+
+            var comp = CreateCompilation(new[] { src, IsExternalInitTypeDefinition }, options: TestOptions.DebugExe);
+            CompileAndVerify(comp, expectedOutput: "C1 { I1 = 42, P2 = 43, P3 = 44 }", verify: Verification.Skipped /* init-only */);
+            comp.VerifyEmitDiagnostics(
+                // (12,32): warning CS0067: The event 'C1.a' is never used
+                //     public event System.Action a;
+                Diagnostic(ErrorCode.WRN_UnreferencedEvent, "a", isSuppressed: false).WithArguments("C1.a").WithLocation(12, 32),
+                // (14,17): warning CS0414: The field 'C1.field1' is assigned but its value is never used
+                //     private int field1 = 100;
+                Diagnostic(ErrorCode.WRN_UnreferencedFieldAssg, "field1", isSuppressed: false).WithArguments("C1.field1").WithLocation(14, 17)
+                );
+        }
+
+        [Fact, WorkItem(47672, "https://github.com/dotnet/roslyn/issues/47672")]
+        public void ToString_PrivateGetter()
+        {
+            var src = @"
+var c1 = new C1();
+System.Console.Write(c1.ToString());
+
+record C1
+{
+    public int P1 { private get => 43; set => throw null; }
+}
+";
+
+            var comp = CreateCompilation(new[] { src, IsExternalInitTypeDefinition }, options: TestOptions.DebugExe);
+            CompileAndVerify(comp, expectedOutput: "C1 { P1 = 43 }", verify: Verification.Skipped /* init-only */);
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact, WorkItem(47797, "https://github.com/dotnet/roslyn/issues/47797")]
+        public void ToString_OverridenProperty_NoRepetition()
+        {
+            var src = @"
+System.Console.WriteLine(new B() { P = 2 }.ToString());
+
+abstract record A
+{
+    public virtual int P { get; set; }
```

Comment:
>public virtual [](start = 4, length = 14)

For completeness, consider testing abstract property as well:
```C#
public abstract int Q { get; set; }
```

Reply:
abstract property is already tested (a few tests above)

Reply:
> abstract property is already tested (a few tests above)

It looks like the duplicated member issue requires explicit properties. See sharplab.io.

---
In reply to: [490564959](https://github.com/dotnet/roslyn/pull/47734#discussion_r490564959) [](ancestors = 490564959)

Reply:
@cston An unrelated question in your sharplab sample. The synthesized clone for `B2` is returning `A2`.

Shouldn't it be returning B2 instead? The spec says that it should use covariant if it's available.

```
    [System.Runtime.CompilerServices.NullableContext(1)]
    public override A2 <Clone>$()
    {
        return new B2(this);
    }
```

And it seems that covariant is available on master:

https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+ABAzAAgmFgQgGMZ9sAmfAIWjgFgAoAb1fy6oKJPMoMocfAGEANgHsAdnAAUASgDcrAL6tWeQsRikKVWgBFEASwBucFKw4tuPfGZMIYAVwjj8xhOctipsxXw2KgB2fGkXcXElfHUWVSA

Am I missing something or this is a bug?

Reply:
@Youssef1313 This is an implementation details. We're tracking this post 16.8: https://github.com/dotnet/roslyn/projects/57#card-40342371